reactive/calico.py

```diff
@@ -100,71 +99,11 @@ def repull_calico_node_image():
     remove_state('calico.service.installed')
 
 
-@when('leadership.is_leader', 'leadership.set.calico-v3-data-migration-needed',
-      'etcd.available', 'calico.etcd-credentials.installed')
-def upgrade_v3_migrate_data():
-    status.maintenance('Migrating data to Calico 3')
-    try:
-        calico_upgrade.configure()
-        calico_upgrade.dry_run()
-        calico_upgrade.start()
-    except Exception:
-        log(traceback.format_exc())
-        message = 'Calico upgrade failed, see debug log'
-        status.blocked(message)
-        return
-    leader_set({'calico-v3-data-migration-needed': None})
-
-
 @when('leadership.is_leader')
-@when_not('leadership.set.calico-v3-data-migration-needed')
 def v3_data_ready():
     leader_set({'calico-v3-data-ready': True})
```

Comment:
I think calico-v3-data-ready could be removed too. It was used to stop certain handlers like `install_calico_binaries` from running until after v2 -> v3 data migration had occurred. It serves no function now.

Reply:
Excellent suggestion. Might even see a perf boost when we stop running the `v3_data_ready` handler every 5 minutes :)